src/codemodder/codemods/base_visitor.py

```diff
@@ -48,7 +50,18 @@ def node_is_selected(self, node) -> bool:
 
     def node_position(self, node):
         # See https://github.com/Instagram/LibCST/blob/main/libcst/_metadata_dependent.py#L112
-        return self.get_metadata(PositionProvider, node)
+        match node:
+            case cst.FunctionDef():
+                # By default a function's position includes the entire
+                # function definition. Instead, we will only use the first line
+                # of the function definition.
+                params_end = self.get_metadata(PositionProvider, node.params).end
+                return CodeRange(
+                    start=self.get_metadata(PositionProvider, node).start,
+                    end=CodePosition(params_end.line, params_end.column + 1),
+                )
```

Comment:
I'd rater have a general solution instead making one for every node type. I suggest changing either includes/excludes match or `node_is_selected` to match match any node that begins on a given line.

Reply:
I'm not sure I agree with a more general solution like that. In that case, why do we care about start/end columns at all right now? Is there not a reason why we do column matching in `match_location` nearby code? If there is, then that more general solution of just matching start line doesn't make much sense.